### DISCUSSION TITLE
Enhancment for SmoothFonts (gradient)

### DISCUSSION THREAD

**Discussion by DrNeuroSurg** [Bodmer/TFT_eSPI#2190] [New processor/display support or capability request]
Great library - ...

There is a function "setCallback(getColorCallback getCol)" - 

as I can see, this callback is only used for background drawing. Hmm.
So I modified (for testing) 

the line
 " **if (getColor) bg = getColor(x + cx, y + cy);**"
to 
" **if (getColor) fg = getColor(x + cx, y + cy);**"

Now I can set a gradient for the TextColor (foreground) of the smooth font.

I think, it's an enhancement to add the possibility for a second (foreground) callback.

![Gradient](https://user-images.githubusercontent.com/3300040/202690792-3fb23132-2d1d-4fdd-a2ec-f530922c9e66.jpg)

**Comment by Bodmer** [comment]
Thank you for the suggestion. It is a nice solution to get vertical gradients in smooth fonts or stepped colour changes, where say top half of character is one colour and bottom half is another.

Another approach is to render the characters into an 8 bit sprite and use it as an alpha mask, then any direction gradient could be applied. Your idea is much simpler to implement though.

I will think on this further
